src/components/actions/ActionBarReducers.ts

```diff
@@ -18,15 +20,23 @@ export const actionBarReducer = (state: IActionBarState = actionBarInitialState,
       if (state.id !== action.payload.id) {
         return state;
       }
-      return {
-        id: state.id,
-        actions: action.payload.actions
-      };
+      return state.id !== action.payload.id
+        ? state
+        : { ...state, actions: action.payload.actions };
     case ActionBarActions.add:
       return {
         id: action.payload.id,
-        actions: []
+        actions: [],
+        isLoading: false,
       };
+    case LoadingActions.turnOn:
```

Comment:
I try to figure out why we need to sent ids in the payload instead of the state array of each action bar if the id is one of then ?

Reply:
we should discuss this in person, I don't understand what you mean

Reply:
@pochretien loading is started by outside components. We usually have a constant with the id of what we want to turn on/off when the page is loading.